HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,61 @@
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+def reducer(tup):
+    left_val = tup[0]
+    right_vals = tup[1]
+    if len(right_vals) == 0:
+        return list(left_val)[0]
+    elif list(left_val)[0] == -1:
+        return list(right_vals)[0]
+    else:
+        return list(left_val)[0]
+
+def divider(tup):
+    dist = tup[0]
+    node_vals = tup[1]
+    return [(nv, dist + 1) for nv in node_vals]
+
+#######
+# Implements Breadth First Search
+# Arguments:
+# 	adj (KV RDD): Adjacency list graph representation. Value is an iterator over neighbors.
+# 	start (string): Where the breadth first search will start
+#   sc: SparkContext
+# Returns:
+# 	(RDD) Distances to each point
```

Comment:
i like your style of comments -- makes it clear what your most important functions are doing and why.